src/include/session.h

```diff
@@ -30,6 +30,28 @@ struct __wt_hazard {
 #endif
 };
 
+/*
+ * WT_HAZARD_WEAK --
+ *	A weak hazard pointer.
+ */
+struct __wt_hazard_weak {
```

Comment:
I feel like we should have a hazard.h but that's not really this PR's issue.

Reply:
I agree - We should get that in as a part of the next few changes.

Reply:
We can include this in a future refactoring. For now, let's leave the existing hazard code alone.

Reply:
Marking as resolved - leaving it for a later refactoring.